pkgs/applications/networking/taler/default.nix

```diff
@@ -1,57 +1,103 @@
-        inherit sha256;
-      };
-      enableParallelBuilding = true;
-      meta = with lib; {
-        description = "Anonymous, taxable payment system.";
-        homepage = "https://taler.net/";
-        license = licenses.agpl3Plus;
-        maintainers = with maintainers; [ ehmry ];
-        platforms = platforms.gnu ++ platforms.linux;
-      };
-    });
-
 in rec {
-
-  taler-exchange = mkTaler {
+  taler-exchange = stdenv.mkDerivation rec {
     pname = "taler-exchange";
-    version = "0.8.1";
-    sha256 = "sha256-MPt3n1JXd0Y89b1qCuF6YxptSr7henfYp97JTq1Z+x4=";
-  } {
+    version = "0.8.5";
+    src = fetchurl {
+      url = "http://ftp.gnu.org/gnu/taler/${pname}-${version}.tar.gz";
+      sha256 = "sha256-jehQKkDYdDkEXen/LSn0gZi4tYpykYM92W35ZbSLNj4=";
+    };
+    postPatch = ''
+      find . -name '*.sh' -exec sed -i 's%#!/bin/bash%#!${stdenv.shell}%g' {} \;
+    '';
+    enableParallelBuilding = true;
+    nativeBuildInputs = [
+      autoreconfHook
+    ];
     buildInputs = [
-      curl
-      jansson
       libgcrypt
       libmicrohttpd
+      jansson
       libsodium
-      libtool
       postgresql
-      # sqlite
+      curl
+      jinja2
+      recutils
+      gettext
+      texinfo # Fix 'makeinfo' is missing on your system.
     ];
     propagatedBuildInputs = [ gnunet' ];
-    patches = [ ./exchange-fix-6665.patch ];
+    configureFlags = [ "--with-gnunet=${gnunet'}" ];
+    checkInputs = [ wget curl ];
+    meta = with lib; {
+      description = ''
+        Taler is an electronic payment system providing the ability to pay
+        anonymously using digital cash.  Taler consists of a network protocol
+        definition (using a RESTful API over HTTP), a Exchange (which creates
+        digital coins), a Wallet (which allows customers to manage, store and
+        spend digital coins), and a Merchant website which allows customers to
+        spend their digital coins.  Naturally, each Merchant is different, but
+        Taler includes code examples to help Merchants integrate Taler as a
+        payment system.
+      '';
+      homepage = "https://taler.net/";
+      license = licenses.agpl3Plus;
+      platforms = platforms.gnu ++ platforms.linux;
+    };
+    # FIXME: at least one test wants to access a postgres db:
+    # `ERROR Database connection to 'postgres:///talercheck' failed: could not connect to server: No such file or directory`
+    doInstallCheck = false;
+    installCheck = ''
+      make check
+    '';
   };
 
-  taler-merchant = mkTaler {
+  taler-merchant = stdenv.mkDerivation rec {
     pname = "taler-merchant";
-    version = "0.8.0";
-    sha256 = "sha256-scrFLXeoQirGqhc+bSQKRl84PfUvjrp1uxF7pfOIB9Q=";
-  } {
-    nativeBuildInputs = [ pkg-config ];
-    buildInputs = taler-exchange.buildInputs ++ [ qrencode taler-exchange ];
+    version = "0.8.3";
+    src = fetchurl {
+      url = "http://ftp.gnu.org/gnu/taler/${pname}-${version}.tar.gz";
```

Comment:
```suggestion
      url = "https://ftp.gnu.org/gnu/taler/${pname}-${version}.tar.gz";
```

Reply:
Same, "mirror://gnu/taler/${pname}-${version}.tar.gz" should work here

Reply:
Using mirror would be actually the right choice here.

Reply:
done